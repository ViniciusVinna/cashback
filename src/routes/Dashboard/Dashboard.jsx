import React, { useState, useRef, useEffect, useCallback } from 'react';
import { useDispatch } from 'react-redux';

import { getPurchases, toggleDialog } from 'actions';

import Waves from 'components/Waves';

import AddPurchase from 'containers/AddPurchase';
import Balance from 'containers/Balance';
import Dialog from 'containers/Dialog';
import Drawer from 'containers/Drawer';
import Profile from 'containers/Profile';
import Transactions from 'containers/Transactions';
import Details from 'containers/Transactions/Details';
import { PurchaseForm } from 'containers/Forms';

import SignInStyled from './Dashboard.styled';

const {
  DashboardPage,
  Content,
  Header,
  HeaderWrapper,
} = SignInStyled;

const Dashboard = () => {
  const [purchaseHeight, setPurchaseHeight] = useState('0px');
  const addPurchaseRef = useRef(null);
  const dispatch = useDispatch();

  const handleClick = useCallback(() => {
    dispatch(toggleDialog(true));
  }, [dispatch]);

  useEffect(() => {
    dispatch(getPurchases());
  }, [dispatch]);

  useEffect(() => {
    const height = addPurchaseRef?.current?.offsetHeight || 0;

    /* istanbul ignore else */
    if (height && height > 0) {
      const paddingHeight = `${height * 2}px`;

      setPurchaseHeight(paddingHeight);
    }
  }, [purchaseHeight, addPurchaseRef]);

  return (
    <DashboardPage
      data-testid="dashboard"
      style={{ paddingBottom: `${purchaseHeight}` }}
    >
      <Header>
        <HeaderWrapper>
          <Profile />
          <Balance />
        </HeaderWrapper>
      </Header>
      <Waves />

      <Content>
        <Transactions />
      </Content>

      <Dialog
        title="Cadastrar Compra"
        renderComponent={(
          <AddPurchase
            onClickHandler={handleClick}
            ref={addPurchaseRef}
          />
        )}
      >
        <PurchaseForm />
      </Dialog>

      <Drawer title="Detalhes da Compra">
        <Details />
      </Drawer>
    </DashboardPage>
  );
};

export default Dashboard;
