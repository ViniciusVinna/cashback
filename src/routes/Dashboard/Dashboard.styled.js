import styled from '@emotion/styled';
import { css } from '@emotion/core';

export default {
  Content: styled('div')(({ theme }) => css`
    padding: 0 ${theme.space.l};
    width: 100%;
    max-width: calc(${theme.breakpoint.xl} /2);
  `),
  DashboardPage: styled('section')(({ theme }) => css`
    align-items: center;
    background-color: ${theme.color.n200};
    display: flex;
    flex-direction: column;
    height: auto;
    justify-content: flex-start;
    min-height: 100vh;
    width: 100vw;

    & > svg {
      max-height: 100px;
      width: 100%;
    }
  `),
  Header: styled('header')(({ theme }) => css`
    background: ${theme.color.g500};
    background: linear-gradient(180deg, ${theme.color.g300} 0%,  ${theme.color.g500} 100%);
    margin-bottom: -1px;
    position: relative;
    width: 100%;
  `),
  HeaderWrapper: styled('div')(({ theme }) => css`
    margin: auto;
    text-align: center;
    width: 100%;
    max-width: calc(${theme.breakpoint.xl} /2);
  `),
};
