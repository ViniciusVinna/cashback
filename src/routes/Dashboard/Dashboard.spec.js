import React from 'react';
import Dashboard from './Dashboard';

describe('Dashboard', () => {
  it('it should render properly', () => {
    const { container } = render(<Dashboard />);
    expect(container).toMatchSnapshot();
  });
});
