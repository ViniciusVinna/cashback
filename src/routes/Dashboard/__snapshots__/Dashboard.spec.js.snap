// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard it should render properly 1`] = `
<div>
  <section
    class="css-2lklxh"
    data-testid="dashboard"
    style="padding-bottom: 0px;"
  >
    <header
      class="css-1s87fvt"
    >
      <div
        class="css-1pe5gt1"
      >
        <div
          class="css-157n9gf"
          data-testid="profile"
        >
          <div
            class="css-1g47b3l"
          >
            <div
              class="css-1mcyu8t"
              data-testid="logo"
            >
              <svg
                preserveAspectRatio="xMidYMid slice"
              >
                sign.svg
              </svg>
            </div>
          </div>
          <div
            class="css-pqvzuk"
          >
            <h6
              class="css-1b7lpw4"
            >
              Vin√≠cius Ribeiro
            </h6>
          </div>
          <figure
            class="css-q5erda"
          >
            <img
              alt="Imagem de perfil"
              src="avatar-2.svg"
            />
          </figure>
        </div>
        <div
          class="css-1dx82la"
          data-testid="balance"
        >
          <header
            class="css-d7tq2t"
          >
            <label
              class="css-1miv8ke"
            >
              Saldo
            </label>
            <div
              class="css-111w3cr"
            >
              R$ 245,00
            </div>
          </header>
          <div
            class="css-1jy0m1q"
          >
            <div
              class="css-gi4i87"
            >
              <label
                class="css-nwvdww"
              >
                Aprovado
              </label>
              <div
                class="css-ybx47f"
              >
                R$ 400,00
              </div>
            </div>
            <div
              class="css-gi4i87"
            >
              <label
                class="css-nwvdww"
              >
                Pendente
              </label>
              <div
                class="css-ybx47f"
              >
                R$ 75,00
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <svg
      viewBox="0 0 1440 320"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0,32L60,58.7C120,85,240,139,360,144C480,149,600,107,720,80C840,53,960,43,1080,74.7C1200,107,1320,181,1380,218.7L1440,256L1440,0L1380,0C1320,0,1200,0,1080,0C960,0,840,0,720,0C600,0,480,0,360,0C240,0,120,0,60,0L0,0Z"
        fill="#52b947"
        fill-opacity="1"
      />
    </svg>
    <div
      class="css-1trrgiq"
    >
      <div
        class="css-4xhxa"
        data-testid="transactions"
      >
        <div
          class="css-1a0ws9n"
          data-transactionid="1"
        >
          <div
            class="css-1nmedbh"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
          </div>
          <div
            class="css-4xhxa"
          >
            <span
              class="css-12kapqf"
            >
              123456789
            </span>
            <div
              class="css-1iqjwv1"
            >
              1600623109000
            </div>
          </div>
          <div
            class="css-iqfsi7"
          >
            R$ 25,00
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </div>
        <div
          class="css-1a0ws9n"
          data-transactionid="2"
        >
          <div
            class="css-1nmedbh"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
          </div>
          <div
            class="css-4xhxa"
          >
            <span
              class="css-12kapqf"
            >
              123456
            </span>
            <div
              class="css-1iqjwv1"
            >
              1600615909000
            </div>
          </div>
          <div
            class="css-iqfsi7"
          >
            R$ 25,00
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </div>
        <div
          class="css-1a0ws9n"
          data-transactionid="3"
        >
          <div
            class="css-1nmedbh"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
          </div>
          <div
            class="css-4xhxa"
          >
            <span
              class="css-12kapqf"
            >
              123456
            </span>
            <div
              class="css-1iqjwv1"
            >
              1600610449000
            </div>
          </div>
          <div
            class="css-iqfsi7"
          >
            R$ 25,00
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </div>
        <div
          class="css-1a0ws9n"
          data-transactionid="4"
        >
          <div
            class="css-1nmedbh"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
          </div>
          <div
            class="css-4xhxa"
          >
            <span
              class="css-12kapqf"
            >
              123456
            </span>
            <div
              class="css-1iqjwv1"
            >
              19/09/2020
            </div>
          </div>
          <div
            class="css-iqfsi7"
          >
            R$ 25,00
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </div>
        <div
          class="css-1a0ws9n"
          data-transactionid="5"
        >
          <div
            class="css-1nmedbh"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
          </div>
          <div
            class="css-4xhxa"
          >
            <span
              class="css-12kapqf"
            >
              123456
            </span>
            <div
              class="css-1iqjwv1"
            >
              1600437649000
            </div>
          </div>
          <div
            class="css-iqfsi7"
          >
            R$ 25,00
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </div>
        <div
          class="css-1a0ws9n"
          data-transactionid="6"
        >
          <div
            class="css-1nmedbh"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
          </div>
          <div
            class="css-4xhxa"
          >
            <span
              class="css-12kapqf"
            >
              123456
            </span>
            <div
              class="css-1iqjwv1"
            >
              1600434049000
            </div>
          </div>
          <div
            class="css-iqfsi7"
          >
            R$ 25,00
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </div>
        <div
          class="css-1a0ws9n"
          data-transactionid="7"
        >
          <div
            class="css-1nmedbh"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
          </div>
          <div
            class="css-4xhxa"
          >
            <span
              class="css-12kapqf"
            >
              123456
            </span>
            <div
              class="css-1iqjwv1"
            >
              1600174849000
            </div>
          </div>
          <div
            class="css-iqfsi7"
          >
            R$ 25,00
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </div>
        <div
          class="css-1a0ws9n"
          data-transactionid="8"
        >
          <div
            class="css-1nmedbh"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
          </div>
          <div
            class="css-4xhxa"
          >
            <span
              class="css-12kapqf"
            >
              123456
            </span>
            <div
              class="css-1iqjwv1"
            >
              1599915649000
            </div>
          </div>
          <div
            class="css-iqfsi7"
          >
            R$ 25,00
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </div>
        <div
          class="css-1a0ws9n"
          data-transactionid="9"
        >
          <div
            class="css-1nmedbh"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
          </div>
          <div
            class="css-4xhxa"
          >
            <span
              class="css-12kapqf"
            >
              123456
            </span>
            <div
              class="css-1iqjwv1"
            >
              1599829249000
            </div>
          </div>
          <div
            class="css-iqfsi7"
          >
            R$ 25,00
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </div>
        <div
          class="css-1a0ws9n"
          data-transactionid="10"
        >
          <div
            class="css-1nmedbh"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
          </div>
          <div
            class="css-4xhxa"
          >
            <span
              class="css-12kapqf"
            >
              123456
            </span>
            <div
              class="css-1iqjwv1"
            >
              1599742849000
            </div>
          </div>
          <div
            class="css-iqfsi7"
          >
            R$ 25,00
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </div>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </div>
        <div
          class="css-celd7p"
          data-testid="drawer"
        >
          <header
            class="css-k5577l"
          >
            <header
              class="css-16zn5ys"
            >
              <svg
                class="css-d4tpuw"
                data-testid="close-drawer"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M427 234.625H167.296l119.702-119.702L256 85 85 256l171 171 29.922-29.924-118.626-119.701H427v-42.75z"
                />
              </svg>
              <h4
                class="css-tzbvbu"
              >
                Detalhes da Compra
              </h4>
            </header>
            <svg
              viewBox="0 0 1440 320"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0,32L60,58.7C120,85,240,139,360,144C480,149,600,107,720,80C840,53,960,43,1080,74.7C1200,107,1320,181,1380,218.7L1440,256L1440,0L1380,0C1320,0,1200,0,1080,0C960,0,840,0,720,0C600,0,480,0,360,0C240,0,120,0,60,0L0,0Z"
                fill="#52b947"
                fill-opacity="1"
              />
            </svg>
          </header>
          <div
            class="css-ypkpmz"
          >
            <div
              class="css-1fotvew"
            >
              <div
                class="css-2zvtfr"
              >
                <div
                  class="css-1cieg4q"
                >
                  <div
                    class="css-m1mwgo"
                    data-testid="transaction-icon"
                    name="id"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M21 2l-2 2m-7.61 7.61a5.5 5.5 0 1 1-7.778 7.778 5.5 5.5 0 0 1 7.777-7.777zm0 0L15.5 7.5m0 0l3 3L22 7l-3-3m-3.5 3.5L19 4"
                      />
                    </svg>
                  </div>
                  Identificador
                </div>
                <div
                  class="css-1duz7hq"
                >
                  1
                </div>
              </div>
              <div
                class="css-2zvtfr"
              >
                <div
                  class="css-1cieg4q"
                >
                  <div
                    class="css-m1mwgo"
                    data-testid="transaction-icon"
                    name="code"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"
                      />
                      <line
                        x1="7"
                        x2="7"
                        y1="7"
                        y2="7"
                      />
                    </svg>
                  </div>
                  C√≥digo
                </div>
                <div
                  class="css-1duz7hq"
                >
                  123456
                </div>
              </div>
              <div
                class="css-2zvtfr"
              >
                <div
                  class="css-1cieg4q"
                >
                  <div
                    class="css-m1mwgo"
                    data-testid="transaction-icon"
                    name="value"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <line
                        x1="12"
                        x2="12"
                        y1="1"
                        y2="23"
                      />
                      <path
                        d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
                      />
                    </svg>
                  </div>
                  Total
                </div>
                <div
                  class="css-1duz7hq"
                >
                  R$ 25,00
                </div>
              </div>
              <div
                class="css-2zvtfr"
              >
                <div
                  class="css-1cieg4q"
                >
                  <div
                    class="css-m1mwgo"
                    data-testid="transaction-icon"
                    name="date"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        height="18"
                        rx="2"
                        ry="2"
                        width="18"
                        x="3"
                        y="4"
                      />
                      <line
                        x1="16"
                        x2="16"
                        y1="2"
                        y2="6"
                      />
                      <line
                        x1="8"
                        x2="8"
                        y1="2"
                        y2="6"
                      />
                      <line
                        x1="3"
                        x2="21"
                        y1="10"
                        y2="10"
                      />
                    </svg>
                  </div>
                  Data
                </div>
                <div
                  class="css-1duz7hq"
                >
                  1600623109000
                </div>
              </div>
              <div
                class="css-2zvtfr"
              >
                <div
                  class="css-1cieg4q"
                >
                  <div
                    class="css-m1mwgo"
                    data-testid="transaction-icon"
                    name="time"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                      />
                      <polyline
                        points="12 6 12 12 16 14"
                      />
                    </svg>
                  </div>
                  Hora
                </div>
                <div
                  class="css-1duz7hq"
                >
                  1600623109000
                </div>
              </div>
              <div
                class="css-2zvtfr"
              >
                <div
                  class="css-1cieg4q"
                >
                  <div
                    class="css-m1mwgo"
                    data-testid="transaction-icon"
                    name="status"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                      />
                      <line
                        x1="12"
                        x2="12"
                        y1="16"
                        y2="12"
                      />
                      <line
                        x1="12"
                        x2="12"
                        y1="8"
                        y2="8"
                      />
                    </svg>
                  </div>
                  Status
                </div>
                <div
                  class="css-1duz7hq"
                >
                  APROVADO
                </div>
              </div>
              <div
                class="css-2zvtfr"
              >
                <div
                  class="css-1cieg4q"
                >
                  <div
                    class="css-m1mwgo"
                    data-testid="transaction-icon"
                    name="cashback"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <polyline
                        points="1 4 1 10 7 10"
                      />
                      <path
                        d="M3.51 15a9 9 0 1 0 2.13-9.36L1 10"
                      />
                    </svg>
                  </div>
                  Cashback
                </div>
                <div
                  class="css-1duz7hq"
                >
                  R$ 5,00
                </div>
              </div>
              <div
                class="css-2zvtfr"
              >
                <div
                  class="css-1cieg4q"
                >
                  <div
                    class="css-m1mwgo"
                    data-testid="transaction-icon"
                    name="percentage"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <line
                        x1="19"
                        x2="5"
                        y1="5"
                        y2="19"
                      />
                      <circle
                        cx="6.5"
                        cy="6.5"
                        r="2.5"
                      />
                      <circle
                        cx="17.5"
                        cy="17.5"
                        r="2.5"
                      />
                    </svg>
                  </div>
                  Porcentagem de cashback
                </div>
                <div
                  class="css-1duz7hq"
                >
                  2%
                </div>
              </div>
              <div
                class="css-2zvtfr"
              >
                <div
                  class="css-1cieg4q"
                >
                  <div
                    class="css-m1mwgo"
                    data-testid="transaction-icon"
                    name="reason"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M21 15a2 2 0 0 1-2 2H7l-4 4V5a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2z"
                      />
                    </svg>
                  </div>
                  Justificativa
                </div>
                <div
                  class="css-1duz7hq"
                >
                  Status aprovado pelo valor da compra
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-ww0bbz"
        />
      </div>
    </div>
    <div
      class="css-5uhkd6"
      data-testid="add-purchase"
    >
      <div
        class="css-172jk2k"
      >
        <button
          class="css-rztmuv"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="5"
              y2="19"
            />
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
          </svg>
          <canvas
            height="0"
            style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            width="0"
          />
        </button>
      </div>
      Cadastrar Compra
    </div>
    <div
      class="css-celd7p"
      data-testid="drawer"
    >
      <header
        class="css-k5577l"
      >
        <header
          class="css-16zn5ys"
        >
          <svg
            class="css-d4tpuw"
            data-testid="close-drawer"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M427 234.625H167.296l119.702-119.702L256 85 85 256l171 171 29.922-29.924-118.626-119.701H427v-42.75z"
            />
          </svg>
          <h4
            class="css-tzbvbu"
          >
            Informa√ß√µes de compra
          </h4>
        </header>
        <svg
          viewBox="0 0 1440 320"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0,32L60,58.7C120,85,240,139,360,144C480,149,600,107,720,80C840,53,960,43,1080,74.7C1200,107,1320,181,1380,218.7L1440,256L1440,0L1380,0C1320,0,1200,0,1080,0C960,0,840,0,720,0C600,0,480,0,360,0C240,0,120,0,60,0L0,0Z"
            fill="#52b947"
            fill-opacity="1"
          />
        </svg>
      </header>
      <div
        class="css-ypkpmz"
      >
        <p>
          Informa√ß√µes de compra
        </p>
      </div>
    </div>
    <div
      class="css-ww0bbz"
    />
  </section>
</div>
`;
